src/pybind/mgr/cephadm/services/smb.py

```diff
@@ -117,6 +120,24 @@ def ignore_possible_stray(
             return True
         return False
 
+    def ok_to_stop(self,
+                   daemon_ids: List[str],
+                   force: bool = False,
+                   known: Optional[List[str]] = None) -> HandleCommandResult:
+        # if only 1 smb, alert user (this is not passable with --force)
+        warn, warn_message = self._enough_daemons_to_stop(self.TYPE, daemon_ids, 'SMB', 1, True)
+        if warn:
+            return HandleCommandResult(-errno.EBUSY, '', warn_message)
+
+        # if reached here, there is > 1 smb daemon.
+        if force:
+            return HandleCommandResult(0, warn_message, '')
+
+        # if reached here, > 1 smb daemon and no force flag.
+        # Provide warning
+        warn_message = "WARNING: Removing SMB daemons can cause clients to lose connectivity. "
+        return HandleCommandResult(-errno.EBUSY, '', warn_message)
+
```

Comment:
please use black formatting in new smb related code. It is not enforced here because of the location of this file

Reply:
> please use black formatting in new smb related code. It is not enforced here because of the location of this file

i did `check-black` and seems OK

Reply:
What I meant by 'It is not enforced here because of the location of this file' also means it is not covered by check-black. Beacause the check-black script works on a package basis and this file is not in the smb package. :-)

Reply:
The formatter in my head sees some issues. :-)

Reply:
ah okay, I see! I just updated patch & tried using black formatting similar to smb code